test/e2e/framework/workspaces.go

```diff
@@ -156,3 +164,21 @@ func NewOrganizationFixture(t *testing.T, server RunningServer, options ...Clust
 	org := NewOrganizationFixtureObject(t, server, options...)
 	return logicalcluster.Name(org.Status.Cluster)
 }
+
+// CreateResources creates all resources from a filesystem in the given workspace identified by upstreamConfig
+func CreateResources(t *testing.T, ctx context.Context, fs embed.FS, upstreamConfig *rest.Config, clusterName logicalcluster.Path, transformers ...helpers.TransformFileFunc) error {
```

Comment:
this has not much to do with workspace, does it?

Please derive ctx from `t`.

Reply:
`test/e2e/framework/util.go` ?

Reply:
> Please derive ctx from t.

`ctx` is typically already derived from t at the test level: https://github.com/kcp-dev/kcp/blob/bbde21cffe43aee7b8f9e78019da6135ea86fb16/test/e2e/reconciler/deployment/deployment_coordinator_test.go#L51-L52

Reply:
Maybe instead we should keep `ctx` and remove the `t` arg from here, and just return the error instead of using `require.NoError` inside this function.

Reply:
all done.